src/main/java/edu/harvard/iq/dataverse/DatasetVersion.java

```diff
@@ -1356,24 +1356,37 @@ public List<DatasetRelPublication> getRelatedPublications() {
-                        if (subField.getDatasetFieldType().getName().equals(DatasetFieldConstant.publicationCitation)) {
-                            String citation = subField.getDisplayValue();
-                            relatedPublication.setText(citation);
-                        }
-
-                        
-                        if (subField.getDatasetFieldType().getName().equals(DatasetFieldConstant.publicationURL)) {
-                            // We have to avoid using subField.getDisplayValue() here - because the DisplayFormatType 
-                            // for this url metadata field is likely set up so that the display value is automatically 
-                            // turned into a clickable HTML HREF block, which we don't want to end in our Schema.org JSON-LD output.
-                            // So we want to use the raw value of the field instead, with 
-                            // minimal HTML sanitation, just in case (this would be done on all URLs in getDisplayValue()).
+                        switch (subField.getDatasetFieldType().getName()) {
+                        case DatasetFieldConstant.publicationCitation:
+                            relatedPublication.setText(subField.getDisplayValue());
+                            break;
+                        case DatasetFieldConstant.publicationURL:
+                            // We have to avoid using subField.getDisplayValue() here - because the DisplayFormatType
+                            // for this url metadata field is likely set up so that the display value is automatically
+                            // turned into a clickable HTML HREF block, which we don't want to end in our Schema.org 
+                            // JSON-LD output. So we want to use the raw value of the field instead, with minimal HTML
+                            // sanitation, just in case (this would be done on all URLs in getDisplayValue()).
                             String url = subField.getValue();
                             if (StringUtils.isBlank(url) || DatasetField.NA_VALUE.equals(url)) {
                                 relatedPublication.setUrl("");
                             } else {
                                 relatedPublication.setUrl(MarkupChecker.sanitizeBasicHTML(url));
                             }
+                            break;
+                        case DatasetFieldConstant.publicationIDType:
+                            // QDR idType has a trailing : now (Aug 2021)
+                            // Get sanitized value without any display modifications
+                            subField.getDatasetFieldType().setDisplayFormat("#VALUE");
+                            relatedPublication.setIdType(subField.getDisplayValue());
+                            break;
+                        case DatasetFieldConstant.publicationIDNumber:
+                            // Get sanitized value without any display modifications
```

Comment:
Sanitized? I see a call to sanitizeBasicHTML above. I'm not sure where any sanitization comes in here.

Reply:
If I'm following the logic correctly, these fields go through https://github.com/IQSS/dataverse/blob/c9ec8595e84a04d04648c3881ba2687b2ca3894d/src/main/java/edu/harvard/iq/dataverse/DatasetFieldValue.java#L103 and get any tags stripped (there shouldn't be any but it appears I can put <a> in the ID number field - it doesn't show on in the display and probably shouldn't go to PID providers that way either). 

I'll remove the mention of sanitize on the publicationIdType as it's a cvv field.

Reply:
Ok, thanks.